frontend/lib/src/components/shared/Toolbar/SharedToolbar.tsx

```diff
@@ -112,24 +61,49 @@ const ToolbarRendererContext =
   createContext<ToolbarRendererContextShape | null>(null)
 ToolbarRendererContext.displayName = "ToolbarRendererContext"
 
-export const ToolbarRenderer: FC<PropsWithChildren> = ({ children }) => {
-  const [toolbarProps] = useRequiredContext(ToolbarContext)
+export const ToolbarRendererContextProvider: FC<PropsWithChildren> = ({
+  children,
+}) => {
   const portalRef = useRef<HTMLDivElement>(null)
 
+  const value = useMemo(() => ({ portalRef }), [portalRef])
```

Comment:
What's the added value from memoizing here? Shouldn't the ref be stable?

Reply:
You're correct the ref is stable. But it's because the type of the context is an object. So if it was written `value={{ portalRef }}`, it would create a new object on every render. I think we could just simplify it to not be an object, remove the memo, and just pass the ref as `value={portalRef}`

Reply:
Ah gotcha, that makes sense.